### PR TITLE
using defaultHTTPClient

### PR DESCRIPTION
**Pull request by djsly**
This allows the http client to use a proper connection pool

### CODE REVIEW

File: authz/providers/azure/rbac/rbac.go [hunk 1]
```diff
@@ -30,9 +30,12 @@ import (
 
 	auth "go.kubeguard.dev/guard/auth/providers/azure"
 	"go.kubeguard.dev/guard/auth/providers/azure/graph"
+
 	"go.kubeguard.dev/guard/authz"
 	authzOpts "go.kubeguard.dev/guard/authz/providers/azure/options"
 
+	"go.kubeguard.dev/guard/util/httpclient"
```
Comment (weinong) [inline authz/providers/azure/rbac/rbac.go:37]: can you align this import with the rest of kubeguard imports?

File: authz/providers/azure/rbac/rbac.go [hunk 2]
```diff
@@ -30,9 +30,12 @@ import (
 
 	auth "go.kubeguard.dev/guard/auth/providers/azure"
 	"go.kubeguard.dev/guard/auth/providers/azure/graph"
+
```
Comment (weinong) [inline authz/providers/azure/rbac/rbac.go:33]: remove unnecessary whiteline

**Comment by weinong** [comment]
LGTM

File: authz/providers/azure/rbac/rbac.go (hunk 1 above)
Comment (djsly) [reply authz/providers/azure/rbac/rbac.go:35]: fixed

File: authz/providers/azure/rbac/rbac.go (hunk 2 above)
Comment (djsly) [reply authz/providers/azure/rbac/rbac.go]: fixed

**Review by weinong — APPROVED** [review]
LGTM

**Review by krdhruva — APPROVED** [review]

**Comment by krdhruva** [comment]
@tamalsaha , can you pls release new version?

**Comment by tamalsaha** [comment]
Done! https://github.com/kubeguard/guard/releases/tag/v0.11.2

**Comment by krdhruva** [comment]
Thanks!

**Pull request merged**